cmdstanpy/install_cmdstan.py

```diff
@@ -196,14 +274,25 @@ def main():
         )
     args = parser.parse_args(sys.argv[1:])
 
-    version = vars(args)['version']
-    if version is None:
-        version = latest_version()
-    print('CmdStan version: {}'.format(version))
+    version = latest_version()
+    if vars(args)['version']:
+        version = vars(args)['version']
+
+    if is_version_available(version):
+        print('Installing CmdStan version: {}'.format(version))
+    else:
+        raise ValueError(
+            'Invalid version requested: {}, cannot install.'.format(version)
+        )
+
+    cmdstan_dir = os.path.expanduser(os.path.join('~', _DOT_CMDSTAN))
```

Comment:
User needs to create `.cmdstan` folder? Should it be the default?

Reply:
> How do we want to handle to logic of folders?

excellent and twisty question.  let's go through the possibilities:

a.  directory `$HOME/.cmdstan` exists - this is what we'd like - all CmdStan wrapper interfaces use same installation location

b. directory `$HOME/.cmdstanpy` exists - keep using it because it's probably got the install we want

c. neither exists - create `$HOME/.cmdstan`

the functions that figure this out are:   install_cmdstan.validate_dir,  and utils.cmdstan_path

the logic in the latter is correct:  https://github.com/stan-dev/cmdstanpy/blob/21a40ac9f6c3f78d7dc30de955434af74dfb8e81/cmdstanpy/utils.py#L200-L206

does this make sense?

I'll take another pass at the install script -  not at all sure it's there yet.

Reply:
Yes, that makes sense

Reply:
revised logic - what do you think?

Reply:
I think it is good now.